go/libraries/doltcore/branch_control/expr_parser_node.md

```diff
@@ -0,0 +1,187 @@
+This document is to show the implementation theory behind the `MatchNode` structure in `expr_parser_node.go`.
```

Comment:
This is a good doc to have, thanks for writing.

No real comments, pretty well explained and clears up some points of confusion I had.